server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/transport/AdmissionControlTransportHandler.java

```diff
@@ -50,16 +54,20 @@ public AdmissionControlTransportHandler(
      */
     @Override
     public void messageReceived(T request, TransportChannel channel, Task task) throws Exception {
-        // intercept all the transport requests here and apply admission control
-        try {
-            // TODO Need to evaluate if we need to apply admission control or not if force Execution is true will update in next PR.
-            this.admissionControlService.applyTransportAdmissionControl(this.action);
-        } catch (final OpenSearchRejectedExecutionException openSearchRejectedExecutionException) {
-            log.warn(openSearchRejectedExecutionException.getMessage());
-            channel.sendResponse(openSearchRejectedExecutionException);
-        } catch (final Exception e) {
-            throw e;
+        // skip admission control if force execution is true
+        if (!this.forceExecution) {
+            // intercept the transport requests here and apply admission control
+            try {
+                this.admissionControlService.applyTransportAdmissionControl(this.action, this.admissionControlActionType);
+                actualHandler.messageReceived(request, channel, task);
+            } catch (final OpenSearchRejectedExecutionException openSearchRejectedExecutionException) {
+                log.warn(openSearchRejectedExecutionException.getMessage());
+                channel.sendResponse(openSearchRejectedExecutionException);
```

Comment:
Can we verify if we are shortcutting here after rejection ? I mean the channel is still open....  Dont we need to close channel ?

Reply:
Yes the channel is still open of we send rejection response. I think we don't need to close the channel for every rejection.